docs/performance/inserts/parallel.rst

```diff
@@ -33,19 +34,31 @@ Example
 =======
 
 Suppose we have a stream of data we want to persist into CrateDB.
-You can parallelize this in Java using a
-`CompletableFuture`_ object, like so:
+You can parallelize this in Java using a `CompletableFuture`_ object in order to
+issue insert statements concurrently (non-blocking).
+
+As the java JDBC `Connection`_ is not thread-safe, each insert must use a
+dedicated connection.
+
+A common solution to avoid over-instantiating of connection objects is the usage
+of a pooled JDBC `DataSource`_ like `HikariCP`_, `vibur-dbcp`_, `commons-dbcp`_
+etc.
+
+In the following example, `HikariCP`_ is used as a `DataSource`_.
 
 .. code-block:: java
 
+    HikariDataSource ds = new HikariDataSource();
+    ds.setJdbcUrl("crate://localhost:5432/doc?user=crate");
+
     IntStream.iterate(0, i -> i + 2)
         .limit(1000)
         .forEach(i -> {
             CompletableFuture<Integer> insertFuture =
```

Comment:
Do you think it would also make sense to include a `CompletableFuture.allOf(..)` to show how to wait for all operations to be finished?